src/Bridge/Graphql/Type/SchemaBuilder.php

```diff
@@ -0,0 +1,298 @@
+use GraphQL\Type\Definition\Type as GraphQLType;
+use GraphQL\Type\Schema;
+use Symfony\Component\Config\Definition\Exception\InvalidTypeException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\PropertyInfo\Type;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Builder of the GraphQL schema.
+ *
+ * @author Raoul Clais <raoul.clais@gmail.com>
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+class SchemaBuilder
+{
+    private $propertyNameCollectionFactory;
+    private $propertyMetadataFactory;
+    private $resourceNameCollectionFactory;
+    private $resourceMetadataFactory;
+    private $collectionDataProvider;
+    private $itemDataProvider;
+    private $normalizer;
+    private $resourceTypesCache = [];
+
+    /**
+     * @param PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory
+     * @param PropertyMetadataFactoryInterface       $propertyMetadataFactory
+     * @param ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory
+     * @param ResourceMetadataFactoryInterface       $resourceMetadataFactory
+     * @param CollectionDataProviderInterface        $collectionDataProvider
+     * @param ItemDataProviderInterface              $itemDataProvider
+     * @param NormalizerInterface                    $normalizer
+     */
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, NormalizerInterface $normalizer)
+    {
+        $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+        $this->resourceNameCollectionFactory = $resourceNameCollectionFactory;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->collectionDataProvider = $collectionDataProvider;
+        $this->itemDataProvider = $itemDataProvider;
+        $this->normalizer = $normalizer;
+    }
+
+    /**
+     * Gets the schema.
+     *
+     * @return Schema
+     */
+    public function getSchema(): Schema
+    {
+        $queryFields = [];
+
+        foreach ($this->resourceNameCollectionFactory->create() as $resource) {
+            $resourceMetadata = $this->resourceMetadataFactory->create($resource);
+
+            $queryFields = array_merge($queryFields, $this->getQueryFields($resource, $resourceMetadata));
+        }
+
+        return new Schema([
+            'query' => new ObjectType([
+                'name' => 'Query',
+                'fields' => $queryFields,
+            ]),
+        ]);
+    }
+
+    /**
+     * Gets the query fields of the schema.
+     *
+     * @param string           $resource
```

Comment:
All the annotations can be remove (already documented by typehints), same everywhere.